zozo-ios-engineer-codecheck/Data/Request/SearchRepositoriesRequest.swift

```diff
@@ -0,0 +1,46 @@
+//
+//  SearchRepositoriesRequest.swift
+//  zozo-ios-engineer-codecheck
+//
+//  Created by yuki.hamada on 2024/06/28.
+//
+
+import Foundation
+
+/// APIRequest に則ってRequestを作成
+///
+/// ```
+/// let request = GIthubAPI.SearchRepository(searchWord: "swift")
+/// ```
+struct SearchRepositoriesRequest: APIRequestProtocol {
+    let searchWord: String
+
+    init(searchWord: String) {
+        self.searchWord = searchWord
+    }
+
+    typealias Response = SearchRepositoriesResponse
+
+    // MARK: - APIRequestProtocol
+
+    var baseURL: URL {
+        guard let baseURL = URL(string: "https://api.github.com") else { fatalError("error baseURL") }
+        return baseURL
+    }
+
+    var path: String {
+        "/search/repositories"
+    }
+
+    var method: HTTPMethod {
+        .get
+    }
+
+    var body: Data? {
+        nil
+    }
```

Comment:
ここをcomputed propertyにしている理由ってあるのかな？ パスやメソッドはこのリクエストの中で変わることはないから定数にしちゃってもいいかなと思う。
```suggestion
    var path = "/search/repositories"
    var method: HTTPMethod = .get
    var body: Data?
```

Reply:
computed property にする必要なかったので、定数へ変更しました！[9361eeb](https://github.com/hamadayuuki/zozo-ios-engineer-codecheck/pull/7/commits/9361eeb12994747f30b1ac5c3c55744489be2d8d)